debug/gdbserver.py

```diff
@@ -833,13 +835,27 @@ def setup(self):
         self.gdb.command("file %s" % self.binary)
 
     def test(self):
-        self.gdb.load()
-        self.gdb.command("b _exit")
+        # Some hart will compete the CRC calculation first!
```

Comment:
Isn't the whole point of GdbSingleHartTest that only one hart actually runs the downloaded program? Is that not happening for some reason?

Reply:
In RTOS mode, when you say "gdb.c", all the harts continue. SO, if you don't give them something to do, they will just continue whatever they were doing. So, if, for real example what they were doing was writing something to 0x8000_0000, the test will fail even though only one hart was running it. SO you still need to park the other harts. 

Am I not understanding why harts would end up parked otherwise?

Reply:
Note that "load" in RTOS mode does NOT set the `$pc` of all harts to `_start`. You have to do it manually if that's what you want.

Reply:
`DownloadTest` is a weird special case because all the harts don't know about the program you created so I can't just tell them to go to `_start`.

Reply:
I'll see if I can make DownloadTest a GdbSingleHartTest. I suspect it's not out of laziness.

Reply:
When I tried it's because they don't know about` _start`, because you create the program on the fly. IIRC.

Thanks for taking a look! That's the last holdout I think...